code/items/ammo/TTTAmmo.cs

```diff
@@ -50,58 +52,83 @@ public override void Spawn()
         {
-            base.Touch(other);
-
-            if (!IsServer || other is not TTTPlayer player)
-            {
-                return;
-            }
+            CurrentAmmo = ammo;
+        }
 
-            string ammoType = AmmoName.ToLower();
-            Inventory inventory = player.Inventory;
+        public override void TakeDamage(DamageInfo info)
+        {
+            PhysicsBody body = info.Body;
 
-            if (!inventory.GetAmmoTypes().Contains(ammoType))
+            if (!body.IsValid())
             {
-                return;
+                body = PhysicsBody;
             }
 
-            int playerAmount = inventory.Ammo.Count(ammoType);
-
-            if (!(Max >= (playerAmount + Math.Ceiling(CurrentAmmo * 0.25))))
+            if (body.IsValid() && !info.Flags.HasFlag(DamageFlags.PhysicsImpact))
             {
-                return;
+                body.ApplyImpulseAt(info.Position, info.Force * 100);
             }
+        }
 
-            int amountGiven = Math.Min(CurrentAmmo, Max - playerAmount);
-            inventory.Ammo.Give(ammoType, amountGiven);
-            CurrentAmmo -= amountGiven;
-            OnPickup.Fire(other);
+        public float HintDistance => 80f;
 
-            if (CurrentAmmo <= 0 || Math.Ceiling(AmmoEntMax * 0.25) > CurrentAmmo)
-            {
-                Delete();
-            }
+        public TranslationData TextOnTick => new("GENERIC_PICKUP", new object[] { Input.GetKeyWithBinding("+iv_use").ToUpper() });
+
+        public bool CanHint(TTTPlayer client)
+        {
+            return true;
         }
 
-        public void SetCurrentAmmo(int ammo)
+        public EntityHintPanel DisplayHint(TTTPlayer client)
         {
-            CurrentAmmo = ammo;
+            return new Hint(TextOnTick);
         }
 
-        public override void TakeDamage(DamageInfo info)
+        public void Tick(TTTPlayer player)
         {
-            PhysicsBody body = info.Body;
+            if (IsClient)
+            {
+                return;
+            }
 
-            if (!body.IsValid())
+            if (player.LifeState != LifeState.Alive)
             {
-                body = PhysicsBody;
+                return;
             }
 
-            if (body.IsValid() && !info.Flags.HasFlag(DamageFlags.PhysicsImpact))
+            using (Prediction.Off())
             {
-                body.ApplyImpulseAt(info.Position, info.Force * 100);
+                if (!Input.Pressed(InputButton.Use))
+                {
+                    return;
+                }
+
+                string ammoType = AmmoName.ToLower();
+                Inventory inventory = player.Inventory;
+
+                if (!inventory.GetAmmoTypes().Contains(ammoType))
+                {
+                    return;
+                }
+
+                int playerAmount = inventory.Ammo.Count(ammoType);
+
+                if (!(Max >= (playerAmount + Math.Ceiling(CurrentAmmo * 0.25))))
```

Comment:
lol dafuq

```suggestion
                if (Max < playerAmount + Math.Ceiling(CurrentAmmo * 0.25))
```

Reply:
Just copying and pasting previous code, we can clean it up